File: implementations/node/sdk/src/contracts/publication/publication-function-upload.ts
import type { AiSdkPublicationCallSite } from "./publication-call-site.js";
import type { AiSdkPublicationDependencyUpload } from "./publication-dependency-upload.js";
import type { AiSdkPublicationFileUpload } from "./publication-file-upload.js";

export interface AiSdkPublicationFunctionUpload {
  readonly site: AiSdkPublicationCallSite;
  readonly environmentRef: string;
  readonly entryPointPath: string;
  readonly entryPointSymbol: string;
  readonly sources?: readonly AiSdkPublicationFileUpload[];
  readonly dependencies?: readonly AiSdkPublicationDependencyUpload[];
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/publication/publication_function_upload.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from .publication_call_site import AiSdkPublicationCallSite
from .publication_dependency_upload import AiSdkPublicationDependencyUpload
from .publication_file_upload import AiSdkPublicationFileUpload


@dataclass(frozen=True)
class AiSdkPublicationFunctionUpload(AiSdkWireModel):
    site: AiSdkPublicationCallSite
    environment_ref: str
    entry_point_path: str
    entry_point_symbol: str
    sources: tuple[AiSdkPublicationFileUpload, ...] = ()
    dependencies: tuple[AiSdkPublicationDependencyUpload, ...] = ()
